controller/telemetry/server.go

```diff
@@ -399,13 +398,24 @@ func convertSampleStream(sample *model.SampleStream) *read.Sample {
 }
 
 func (s *server) requestLabelsFor(requestScope *write.RequestScope) prometheus.Labels {
-	sourceName, sourceDeployment := s.getNameAndDeployment(requestScope.Ctx.SourceIp)
-	targetName, targetDeployment := s.getNameAndDeployment(requestScope.Ctx.TargetAddr.Ip)
+	var sourceDeployment string
+	sourceDeploymentPtr, _ := s.getDeployment(requestScope.Ctx.SourceIp)
```

Comment:
if we always discard the error then why have `getDeployment` return an error at all?

Reply:
due to a poorly-conceived attempt at forwards-compatibility. simplified in 62f7927412ce46231eca67ad9b00103fe7777625